### PR TITLE
MEPTS-1404 | LISTA DE DAH DIFF CCS MAV BD

### PR DESCRIPTION
**Pull request by helderjosue**

### CODE REVIEW

**Review by ItsMurumba — APPROVED** [review]
LGTM

**Review by ItsMurumba — CHANGES_REQUESTED** [review]
Please fix the conflict @helderjosue

**Review by ItsMurumba — APPROVED** [review]
LGTM

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java [hunk 1]
```diff
@@ -151,11 +151,43 @@ public String getCd4ResultBellowOneYear() {
...
+        + "                      ON e.encounter_id = o.encounter_id  "
+        + "INNER JOIN ( "
+        + getLastCd4ResultDateQueries()
+        + " ) last_cd4 ON last_cd4.person_id = ps.person_id  "
+        + " WHERE  ps.voided = 0  "
+        + "  AND e.voided = 0   "
+        + "  AND o.voided = 0  "
+        + "  AND e.encounter_type IN ( ${6}, ${13}, ${51} )  "
+        + "  AND o.concept_id = ${1695}  "
+        + "  AND o.value_numeric < 200  "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java]: Sugest you use the same method and use parameters to control what you want 
@helderjosue

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java [hunk 2]
```diff
@@ -165,12 +197,44 @@ public String getCd4ResultOverOrEqualTo5y() {
...
+        + "                      ON e.encounter_id = o.encounter_id  "
+        + "INNER JOIN ( "
+        + getLastCd4ResultDateQueries()
+        + " ) last_cd4 ON last_cd4.person_id = ps.person_id  "
+        + " WHERE  ps.voided = 0  "
+        + "  AND e.voided = 0   "
+        + "  AND o.voided = 0  "
+        + "  AND e.encounter_type IN ( ${6}, ${13}, ${51} )  "
+        + "  AND o.concept_id = ${1695}  "
+        + "  AND o.value_numeric < 500  "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java]: Here

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java [hunk 3]
```diff
@@ -181,11 +245,44 @@ public String getCd4ResultBetweenOneAnd5y() {
...
+        + "                      ON e.encounter_id = o.encounter_id  "
+        + "INNER JOIN ( "
+        + getLastCd4ResultDateQueries()
+        + " ) last_cd4 ON last_cd4.person_id = ps.person_id  "
+        + " WHERE  ps.voided = 0  "
+        + "  AND e.voided = 0   "
+        + "  AND o.voided = 0  "
+        + "  AND e.encounter_type IN ( ${6}, ${13}, ${51} )  "
+        + "  AND o.concept_id = ${1695}  "
+        + "  AND o.value_numeric < 750  "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java]: here

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@helderjosue have a look

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java (hunk 1 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java]: Implemented

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java (hunk 2 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java]: done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java (hunk 3 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/advancedhivillness/ListOfPatientsOnAdvancedHivIllnessQueries.java]: done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupListOfPatientsInAdvancedHivIllness.java [hunk 4]
```diff
@@ -16,8 +16,9 @@
 import org.openmrs.module.reporting.report.ReportDesign;
 import org.openmrs.module.reporting.report.definition.ReportDefinition;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
-@Deprecated
+@Component
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupListOfPatientsInAdvancedHivIllness.java]: Why?

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**